stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml

```diff
@@ -89,10 +92,49 @@
         >
             <attribute name="t-if">move_line.id</attribute>
         </xpath>
-        <!-- Remove "All items could not be shipped..." sentence -->
+        <!-- Remove "All items could not be shipped..." sentence
+            Substitute it by a message indicating if all goods have been delivered. -->
         <xpath expr='//div[hasclass("page")]/p[last()]' position="attributes">
             <attribute name="t-if">False</attribute>
         </xpath>
+        <xpath expr='//div[hasclass("page")]/p[last()]' position="after">
+            <t t-if="o.picking_type_id.code == 'outgoing'">
+                <t t-set="remainings" t-value="get_remaining_to_deliver(o)" />
+                <t t-if="not remainings">
+                    <p
+                        class="remaining-to-deliver-separator"
+                    >All ordered items have been delivered.</p>
+                </t>
+                <t t-else="">
+                    <p class="remaining-to-deliver-separator">Remaining to deliver:</p>
```

Comment:
you could simply use an `bg-dark` class. See https://getbootstrap.com/docs/4.0/utilities/colors/